### PR TITLE
Base API for notes

### PR DESCRIPTION
**Pull request by krisx303**

### CODE REVIEW

File: src/main/java/com/brightpath/learnify/controller/NotesController.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    private final User user;
+
+    public NotesController(NoteService notesService, UserService userService) {
+        this.notesService = notesService;
+        this.user = userService.createUser("test@user.com", "Test User");
+    }
+
+    @Override
+    public ResponseEntity<NoteSummaryDto> getNoteById(String noteId) {
+        var note = notesService.getNoteById(UUID.fromString(noteId));
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/controller/NotesController.java]: what type is it?

File: src/main/java/com/brightpath/learnify/persistance/note/Note.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.brightpath.learnify.persistance.note;
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/persistance/note/Note.java]: change package

File: src/main/java/com/brightpath/learnify/persistance/common/Workspace.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.brightpath.learnify.persistance.common;
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/persistance/common/Workspace.java:1]: change package

File: src/main/java/com/brightpath/learnify/controller/NotesController.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        return ResponseEntity.ok(asNoteSummaryDto(note));
+    }
+
+    @Override
+    public ResponseEntity<NoteSummaryDto> createNote(NoteCreateDto noteCreateDto) {
+        var note = notesService.createNote(
+                noteCreateDto.getTitle(),
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/controller/NotesController.java:40]: maybe create constructor in Note?

File: src/main/java/com/brightpath/learnify/controller/NotesController.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        return ResponseEntity.ok("Note updated");
+    }
+
+    @Override
+    public ResponseEntity<String> getNoteContentPage(String noteId, Integer pageNumber) {
+        var content = notesService.getNoteContentPage(UUID.fromString(noteId), pageNumber);
+        return ResponseEntity.ok(content);
+    }
+
+    private NoteSummaryDto asNoteSummaryDto(Note note) {
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/controller/NotesController.java:68]: consider using builder

File: src/main/java/com/brightpath/learnify/controller/WorkspaceController.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+public class WorkspaceController implements WorkspacesApi {
+
+    private final WorkspaceService workspaceService;
+
+    public WorkspaceController(WorkspaceService workspaceService) {
+        this.workspaceService = workspaceService;
+        this.init();
+    }
+
+    private void init() {
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/controller/WorkspaceController.java]: can we extract this to different class

File: src/main/java/com/brightpath/learnify/controller/NotesController.java (hunk 5 above)
Comment (krisx303) [reply src/main/java/com/brightpath/learnify/controller/NotesController.java:68]: After simple research I do not think it is possible to add builders for those generated classes automatically

**Review by Adam3004 — APPROVED** [review]

**Pull request merged**